### PR TITLE
allow admins to update node deployment

### LINKED ISSUE #5258: Allow admins to update node deployment
**User Story**
Allow admins to update node deployment

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: allow admins to update node deployment
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5258

```release-note
allow admins to update node deployment
```

### CODE REVIEW

File: api/pkg/handler/v1/node/node.go [hunk 1]
```diff
@@ -631,26 +631,26 @@ func DecodePatchNodeDeployment(c context.Context, r *http.Request) (interface{},
 	return req, nil
 }
 
-func PatchNodeDeployment(sshKeyProvider provider.SSHKeyProvider, projectProvider provider.ProjectProvider, seedsGetter provider.SeedsGetter, userInfoGetter provider.UserInfoGetter) endpoint.Endpoint {
+func PatchNodeDeployment(sshKeyProvider provider.SSHKeyProvider, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, seedsGetter provider.SeedsGetter, userInfoGetter provider.UserInfoGetter) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
 		req := request.(patchNodeDeploymentReq)
 		clusterProvider := ctx.Value(middleware.ClusterProviderContextKey).(provider.ClusterProvider)
-		userInfo, err := userInfoGetter(ctx, req.ProjectID)
+		userInfo, err := userInfoGetter(ctx, "")
```
Comment (maciaszczykm) [inline api/pkg/handler/v1/node/node.go:638]: Does it mean that we are not checking if the user belongs to the project at this point?

Comment (zreigz) [reply api/pkg/handler/v1/node/node.go:638]: it's checked in `common.GetProject` and `cluster.GetCluster` methods which are prepared for admin and regular user

Comment (maciaszczykm) [reply api/pkg/handler/v1/node/node.go:638]: I'd keep it here too but it's not that important at this point.

**Pull request merged**